### PR TITLE
ROC-5133 Under-18 defendant hand-off page

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### DO NOT MERGE!

Waiting on approval from Justice Birrs before we can deploy this

### JIRA link
https://tools.hmcts.net/jira/browse/ROC-5133

### Change description
Show new hand-off page when a defendant declares themself under 18

### Work checklist

- [ ] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [x] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/response/views/eligibility/under-18.njk [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+{% set heading = 'Contact the court' %}
+
+{% block content %}
+
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <p>{{ t('If you are under 18 you need to email or phone the court. Claimants cannot submit a claim against someone under 18 so you may be able to have the claim stopped.') }}</p>
+
+      <p>
+        <strong>{{ t('Money Claim Online helpdesk') }}</strong><br/>
```
Comment (sabahirfan) [inline src/main/features/response/views/eligibility/under-18.njk]: We have few support macros, please check if we can reuse any of them.

Comment (andrewwa-kainos) [reply src/main/features/response/views/eligibility/under-18.njk]: Clarifying with Pete as the contact details are a little different, but great idea! I didn't know about those widgets.

Comment (andrewwa-kainos) [reply src/main/features/response/views/eligibility/under-18.njk]: Marc confirmed these details should be the same as is already present on the mcol eligibility page (and therefore in the macro)

Comment (sabahirfan) [reply src/main/features/response/views/eligibility/under-18.njk]: Thanks 👍

File: src/main/features/response/routes/your-dob.ts [hunk 2]
```diff
@@ -38,11 +40,14 @@ export default express.Router()
     FormValidator.requestHandler(DateOfBirth, DateOfBirth.fromObject),
     ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
       const form: Form<DateOfBirth> = req.body
-
+      const claim: Claim = res.locals.claim
       if (form.hasErrors()) {
-        renderView(form, res)
+        if (form.errors.length === 1 && form.errors[0].message.match(dateUnder18Pattern)) {
```
[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/main/features/response/routes/your-dob.ts]: At the moment I don't think it's possible to have more than one error, true. But separation of concerns: this unit doesn't need to know that. The logic I'm trying to express is "if the only validation error is that the date is too recent, go to the hand-off page". This should cover any future changes too.

Let me know if you still want me to change it.

File: src/main/features/response/views/eligibility/under-18.njk [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+{% set heading = 'Contact the court' %}
+
+{% block content %}
+
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <p>{{ t('If you are under 18 you need to email or phone the court. Claimants cannot submit a claim against someone under 18 so you may be able to have the claim stopped.') }}</p>
+
+      {{ mcolSupportWidget(label='') }}
```
[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/main/features/response/views/eligibility/under-18.njk]: I noticed there were different contact details in different pages and Marc said they should be the same as on https://hmcts-cmc-prototype.herokuapp.com/prototypes/prototype-feb-2019/dashboard/new-claim/eligibility/mcol-eligibility. Want to note that it does seem weird to me, too!

File: src/main/features/response/routes/your-dob.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (robertparkinson) [reply src/main/features/response/routes/your-dob.ts]: I would argue that once the error is under 18 we should go to the hand off page so maybe switch it to a filter to check if that error is in the array?

[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/main/features/response/routes/your-dob.ts]: Agreed

**Pull request merged**